Panic occurs when refreshing the state of `aws_ec2_traffic_mirror_filter_rule` resource

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform CLI and Terraform AWS Provider Version

* terraform: v1.1.2
* terraform-provider-aws: v3.69.0 and v3.70.0 
   * I tried with v3.70.0, but it was also failed. 

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* aws_ec2_traffic_mirror_filter_rule

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

It seems to occur panic when refleshing the state.
Therefore, only the settings related to `aws_ec2_traffic_mirror_filter_rule` are described in this issue.

```hcl
resource "aws_ec2_traffic_mirror_filter" "arkime" {
  description = "arkime"

  tags = {
    Name  = "arkime-staging"
  }
}

resource "aws_ec2_traffic_mirror_filter_rule" "inbound" {
  description              = "only TCP"
  traffic_mirror_filter_id = aws_ec2_traffic_mirror_filter.arkime.id
  destination_cidr_block   = "0.0.0.0/0"
  source_cidr_block        = "0.0.0.0/0"
  destination_port_range {
    from_port = 443
    to_port   = 443
  }
  protocol          = 6
  rule_action       = "accept"
  rule_number       = 1
  traffic_direction = "ingress"
}
```

### Debug Output

https://gist.github.com/44smkn/c0ff8bddab345a1e354fe06510686b8a

### Panic Output

```
Stack trace from the terraform-provider-aws_v3.69.0_x5 plugin:

panic: reflect: call of reflect.Value.IsNil on array Value

goroutine 341 [running]:
reflect.Value.IsNil(...)
    /opt/goenv/versions/1.16.0/src/reflect/value.go:1086
github.com/mitchellh/mapstructure.(*Decoder).decodeSlice(0xc00012d8c8, 0x0, 0x0, 0x751ba60, 0xc001719d50, 0x7356c80, 0xc001692240, 0x197, 0x8, 0x8)
    /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/mitchellh/mapstructure@v1.4.2/mapstructure.go:1095 +0x10ad
github.com/mitchellh/mapstructure.(*Decoder).decode(0xc00012d8c8, 0x0, 0x0, 0x751ba60, 0xc001719d50, 0x7356c80, 0xc001692240, 0x197, 0xc001692240, 0x197)
    /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/mitchellh/mapstructure@v1.4.2/mapstructure.go:469 +0x836
github.com/mitchellh/mapstructure.(*Decoder).Decode(0xc00012d8c8, 0x751ba60, 0xc001719d50, 0x0, 0xc001692240)
    /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/mitchellh/mapstructure@v1.4.2/mapstructure.go:398 +0xf0
github.com/mitchellh/mapstructure.Decode(0x751ba60, 0xc001719d50, 0x72be320, 0xc001692240, 0x1, 0x943db75)
    /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/mitchellh/mapstructure@v1.4.2/mapstructure.go:302 +0xa5
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).setList(0xc0027e1ed8, 0xc001719d60, 0x1, 0x1, 0x751ba60, 0xc001719d50, 0xc000632140, 0x943db75, 0xc001721a40)
    /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/field_writer_map.go:127 +0xd2
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).set(0xc0027e1ed8, 0xc001719d60, 0x1, 0x1, 0x751ba60, 0xc001719d50, 0x1, 0x1)
    /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/field_writer_map.go:103 +0xde
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).WriteField(0xc0027e1ed8, 0xc001719d60, 0x1, 0x1, 0x751ba60, 0xc001719d50, 0x0, 0x0)
    /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/field_writer_map.go:89 +0x3f9
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*ResourceData).Set(0xc002615500, 0x943db75, 0x16, 0x751ba60, 0xc001719d50, 0x0, 0x0)
    /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/resource_data.go:227 +0x1df
github.com/hashicorp/terraform-provider-aws/internal/service/ec2.resourceTrafficMirrorFilterRuleRead(0xc002615500, 0x80eb5a0, 0xc00033a000, 0xee81f80, 0xc000088800)
    /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-aws/internal/service/ec2/traffic_mirror_filter_rule.go:191 +0x5dd
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc0005ada40, 0xa356e48, 0xc002806d00, 0xc002615500, 0x80eb5a0, 0xc00033a000, 0x0, 0x0, 0x0)
    /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/resource.go:346 +0x1ee
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc0005ada40, 0xa356e48, 0xc002806d00, 0xc002619110, 0x80eb5a0, 0xc00033a000, 0xc002669700, 0x0, 0x0, 0x0)
    /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/resource.go:635 +0x1cb
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc000122f30, 0xa356e48, 0xc002806d00, 0xc002806d40, 0xc002806d00, 0x40b965, 0x8d4fba0)
    /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/grpc_provider.go:576 +0x47d
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc002039220, 0xa356ef0, 0xc002806d00, 0xc002803740, 0xc002039220, 0xc000a5fba0, 0x46d147)
    /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.4.0/tfprotov5/tf5server/server.go:298 +0x105
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler(0x912f180, 0xc002039220, 0xa356ef0, 0xc002851a40, 0xc0028036e0, 0x0, 0xa356ef0, 0xc002851a40, 0xc0027fc1e0, 0x1ca)
    /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.4.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:344 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000cfc00, 0xa383478, 0xc002076300, 0xc002600d00, 0xc000a30ab0, 0xee3fcb0, 0x0, 0x0, 0x0)
    /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:1194 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc0000cfc00, 0xa383478, 0xc002076300, 0xc002600d00, 0x0)
    /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:1517 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0018b2f10, 0xc0000cfc00, 0xa383478, 0xc002076300, 0xc002600d00)
    /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:859 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
    /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:857 +0x1fd
```

<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->

### Expected Behavior

The terraform plan completes normally without panic.

### Actual Behavior

The panic occurred when running terraform plan.

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform plan`

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

* #0000

I could reproduce the issue on the main branch.

```console
$ make testacc TESTS=TestAccEC2TrafficMirrorFilterRule PKG=ec2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/ec2/... -v -count 1 -parallel 20 -run='TestAccEC2TrafficMirrorFilterRule' -timeout 180m
=== RUN   TestAccEC2TrafficMirrorFilterRule_basic
=== PAUSE TestAccEC2TrafficMirrorFilterRule_basic
=== RUN   TestAccEC2TrafficMirrorFilterRule_disappears
=== PAUSE TestAccEC2TrafficMirrorFilterRule_disappears
=== CONT  TestAccEC2TrafficMirrorFilterRule_basic
=== CONT  TestAccEC2TrafficMirrorFilterRule_disappears
--- PASS: TestAccEC2TrafficMirrorFilterRule_disappears (54.68s)
panic: reflect: call of reflect.Value.IsNil on array Value

goroutine 1859 [running]:
reflect.Value.IsNil(...)
	/usr/local/Cellar/go/1.17.5/libexec/src/reflect/value.go:1427
github.com/mitchellh/mapstructure.(*Decoder).decodeSlice(0xc003ac90d8, {0x0, 0x0}, {0x73de360, 0xc0024450e0}, {0x722c260, 0xc0017f82e8, 0x203000})
	/Users/shunsuke-suzuki/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.2/mapstructure.go:1095 +0xbfe
github.com/mitchellh/mapstructure.(*Decoder).decode(0xc003ac90d8, {0x0, 0xc00242d901}, {0x73de360, 0xc0024450e0}, {0x722c260, 0xc0017f82e8, 0x48})
	/Users/shunsuke-suzuki/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.2/mapstructure.go:469 +0x3ff
github.com/mitchellh/mapstructure.(*Decoder).Decode(0xc003ac90d8, {0x73de360, 0xc0024450e0})
	/Users/shunsuke-suzuki/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.2/mapstructure.go:398 +0xd8
github.com/mitchellh/mapstructure.Decode({0x73de360, 0xc0024450e0}, {0x719fc40, 0xc0017f82e8})
	/Users/shunsuke-suzuki/go/pkg/mod/github.com/mitchellh/mapstructure@v1.4.2/mapstructure.go:302 +0x91
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).setList(0xc002b2f500, {0xc0024450f0, 0x1, 0x1}, {0x73de360, 0xc0024450e0}, 0x92c8f90)
	/Users/shunsuke-suzuki/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/field_writer_map.go:127 +0xba
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).set(0xc002b2f500, {0xc0024450f0, 0x1, 0x1}, {0x73de360, 0xc0024450e0})
	/Users/shunsuke-suzuki/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/field_writer_map.go:103 +0xa5
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).WriteField(0xc002b2f500, {0xc0024450f0, 0x1, 0x1}, {0x73de360, 0xc0024450e0})
	/Users/shunsuke-suzuki/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/field_writer_map.go:89 +0x3f9
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*ResourceData).Set(0xc002926680, {0x92c8f90, 0x16}, {0x73de360, 0xc0024450e0})
	/Users/shunsuke-suzuki/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/resource_data.go:227 +0x21c
github.com/hashicorp/terraform-provider-aws/internal/service/ec2.resourceTrafficMirrorFilterRuleRead(0xc002926680, {0x7fa95c0, 0xc000943500})
	/Users/shunsuke-suzuki/repos/src/github.com/suzuki-shunsuke/terraform-provider-aws/internal/service/ec2/traffic_mirror_filter_rule.go:191 +0x5c5
github.com/hashicorp/terraform-provider-aws/internal/service/ec2.resourceTrafficMirrorFilterRuleUpdate(0xc002926680, {0x7fa95c0, 0xc000943500})
	/Users/shunsuke-suzuki/repos/src/github.com/suzuki-shunsuke/terraform-provider-aws/internal/service/ec2/traffic_mirror_filter_rule.go:318 +0xfc5
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0x7d07780, {0xa2196d8, 0xc002098e80}, 0x24, {0x7fa95c0, 0xc000943500})
	/Users/shunsuke-suzuki/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/resource.go:363 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0007bce00, {0xa2196d8, 0xc002098e80}, 0xc00237b520, 0xc002926500, {0x7fa95c0, 0xc000943500})
	/Users/shunsuke-suzuki/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/resource.go:475 +0x6ba
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0015f24e0, {0xa2196d8, 0xc002098e80}, 0xc00222b3b0)
	/Users/shunsuke-suzuki/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/grpc_provider.go:977 +0xd8a
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0020ae280, {0xa219780, 0xc00223b860}, 0xc002228380)
	/Users/shunsuke-suzuki/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:603 +0x30e
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x9016380, 0xc0020ae280}, {0xa219780, 0xc00223b860}, 0xc002274a80, 0x0)
	/Users/shunsuke-suzuki/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0009b9500, {0xa30b0e8, 0xc001618180}, 0xc002c51f00, 0xc002096d20, 0xf7b8ee0, 0x0)
	/Users/shunsuke-suzuki/go/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:1194 +0xc8f
google.golang.org/grpc.(*Server).handleStream(0xc0009b9500, {0xa30b0e8, 0xc001618180}, 0xc002c51f00, 0x0)
	/Users/shunsuke-suzuki/go/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:1517 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/Users/shunsuke-suzuki/go/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:859 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/Users/shunsuke-suzuki/go/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:857 +0x294
FAIL	github.com/hashicorp/terraform-provider-aws/internal/service/ec2	67.255s
FAIL
make: *** [testacc] Error 1
```

--- a/.changelog/22315.txt
+++ b/.changelog/22315.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/aws_ec2_traffic_mirror_filter_rule: Prevent crash during resource read
+```

--- a/internal/service/ec2/traffic_mirror_filter_rule.go
+++ b/internal/service/ec2/traffic_mirror_filter_rule.go
@@ -356,7 +356,7 @@ func buildTrafficMirrorFilterRulePortRangeSchema(portRange *ec2.TrafficMirrorPor
 		return nil
 	}
 
-	var out [1]interface{}
+	out := make([]interface{}, 1)
 	elem := make(map[string]interface{})
 	elem["from_port"] = portRange.FromPort
 	elem["to_port"] = portRange.ToPort

